canonical-web-and-design/ubuntu.com
# Fix broken cypress test

## Done
- Updated file name from `_cube-contact-us-form.html` to `_credentialing-contact-us-form.html` as cube pages have been updated to credentialing  by this [PR](https://github.com/canonical-web-and-design/ubuntu.com/pull/11537)
- Updated returnURL from `/cube/thank-you` to `/credentialing/thank-you` in `/credentialing/contact-us.html`
- Updated URL from `/cube/contact-us` to `/credentialing/contact-us` in `/tests/cypress/forms/forms_spec.js` for cypress test
- Wrote seperate cypress test for `/automotive` page as the form in the `/automotive` page has been updated by this [PR](https://github.com/canonical-web-and-design/ubuntu.com/pull/11636)  

## QA

How to run cypress test locally for forms on ubuntu.com 
* Make sure you have marketo keys in your `.env.local` - please contact me if you don't have it.

1. Move this file `/tests/cypress/forms/forms_spec.js` into `/tests/cypress/integration` folder
(Cypress test for forms is only run one time a day so it is in the separated folder called `forms`. The other cypress tests are run every time a PR is made so they are in the `integration` folder.)

2. Run ubuntu.com locally 
3. Go to` package.json` and update `baseURL` of the “cypress:open” script to the url the ubuntu.com is currently running in your machine. i.g.  "cypress:open": "cypress open --config-file tests/cypress/cypress.json --config baseUrl=http://192.168.64.4:${PORT:-8001}/" 
4. `yarn add cypress@8.7.0` if you don’t have cypress installed locally 
5. `yarn run cypress:open` in your terminal 
6. Click `forms_spec.js` 

![Screenshot 2022-06-06 at 11 34 56 am](https://user-images.githubusercontent.com/57550290/172146338-16cf8c7f-4942-487d-8ce2-3ffbe80d7eae.png)
